readme.md

```diff
@@ -178,7 +186,45 @@ specific types and there are rules as to how to spell them.
 	to mean	"scale the number by 1024", "by 1024x1024", "by 1024x1024x1024" or "by 1024x1024x1024x1024"
 	respectively. The suffix is case insensitive, and can also include the `b`, as in `kb` or `MB`.
 
+### Array of complex objects
 
+Creation of more complex objects in array is possible via subsections. Lets say that we have following configuration object:
+
+```csharp
+public class WatchedProcess
+{
+    public string Name { get; set; }
+    public string ApplicationPath { get; set; }
+}
+```
+
+We would like to retrieve from our configuration as `IList<WatchedProcess>`. Even if git-config syntax does not have direct support for this scenario, we are able to create list of complex object with subsection and [ConfigurationRoot](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.configurationroot?view=dotnet-plat-ext-6.0).
+
+`ConfigurationRoot` work with array in way, where it create indicies for items as "subsections". This allow us to create section selector that pick value from array based on index - i.e. `WatchedProcess:0:Name` select value of `Name` in first item in array. With this information in mind we are able to use indicies as subsections and create array of complex objects in following way:
```

Comment:
Slight reword:

`ConfigurationRoot` supports working with arrays by creating indices for items as "subsections". This allows us to create a section selector that picks values from the array based on the element index. For example, `WatchedProcess:0:Name` selects the value `Name` from the first item in the array. This means we can use indices as subsections and create an array of complex objects as follows:

Reply:
FYI: in the future, you could use the suggestion feature :) 
![image](https://user-images.githubusercontent.com/8254631/196696682-762848ea-95d0-4bc6-961a-2d9664404863.png)